tests/components/http/test_ban.py

```diff
@@ -174,9 +176,10 @@ async def unauth_handler(request):
 
         assert len(notification_calls) == 3
         assert (
-            "Login attempt or request with invalid authentication from example.com (200.201.202.204) (Python"
+            "Login attempt or request with invalid authentication from example.com (200.201.202.204) (Test Client 1.0/T1"
```

Comment:
I was leaving this in the original flexible way, so that it is not sensitive to the exact sanitation method used in case we want to change this later.  This is supposed to follow the original test with the additional check that the majority of the user-agent survives. The assertion below that verifies that "<script>" was sanitized (by making sure that its not in the code any more)

We could adopt this to enforcing the exact outcome if you prefer.